package.json

```diff
@@ -41,8 +41,7 @@
 		"lint-staged": "13.2.1",
 		"npm-run-all": "^4.1.5",
 		"prettier": "3.0.3",
-		"tslib": "2.5.3",
-		"typescript": "5.1.3"
```

Comment:
Does TypeScript need to be removed from the top-level package? Does this not increase the risk of mismatched TS versions? And why would `tslib` be left there?

Reply:
> Does TypeScript need to be removed from the top-level package? Does this not increase the risk of mismatched TS versions?

So nothing in the root actually uses TypeScript. 

Right now, this root dependency is only there to provide DCR with TypeScript (because DCR doesn't declare a dependency on it), but not AR (which has a declared dependency an earlier version and is not part of the workspace). 

As you suggest, there are two ways of providing common deps like this - installing them once in the root, or by requiring projects in the workspace to be explicit. 

In csnx, we've found it a lot more straightforward to have projects declare their own dependencies (the ones declared [here](https://github.com/guardian/csnx/blob/main/package.json#L70-L75) are for deps that are also _used_ by the root, so we know are present – this set up also feels a bit weird to me). 

But if you chose to provide them all from the root, then you need to know exactly which dependencies all projects have, manually decide which of those make sense to pull up to the root, and _then_ keep on top of whether those projects continue to need them (since they're not declared anywhere, how do you know?). 

You are also bound to change all projects simultaneously, which can make a simple bump in one package a far more complicated change as you fix a load of issues in unrelated packages.

The trade off is needing to manually keep common dependencies in sync (note these are not peerDeps, just common deps – as long as packages declare `peerDependencies`, this should be safe). 

Moving common dependency declarations to the root does guarantee version parity across projects, but at the cost of flexibility and an increased maintenance overhead (are the assumptions in the root still correct?). 

It's also worth considering that AR and DCR currently use different versions of TypeScript without issue. In CSNX we have linting scripts, for example, in the root that run across all projects that we therefore _want_ to only use the root version of `eslint`. 

I think it's tempting to think that version parity is always preferable over drift, but the cost of forcing it is high and where it's not a direct a benefit (i.e. the code can run) it's not proven to be worth it, at least in our experience of CSNX.

Reply:
> And why would `tslib` be left there?

it's a peer dep of `@guardian/prettier`